website/en/docs/lang/subtypes.md

```diff
@@ -136,3 +136,37 @@ type Func2 = (number, string) => void;
 let func1: Func1 = (a: number) => {};
 let func2: Func2 = func1;
 ```
+
+#### Subtypes of unions of functions <a class="toc" id="toc-subtypes-of-unions-of-functions" href="#toc-subtypes-of-unions-of-functions"></a>
+
+A function is a subtype of another when the input is contravariant (_less specific type / supertype_) and the output is covariant (_more specific type / subtype_).
+
+```js
+// @flow
+declare var foo1: (string => ?string) => void
+declare var bar1: ?string => string
+
+foo1(bar1) // Works!
+```
+So what would be a subtype of a union of functions? Well we know that a subtype of ``A | B`` woudld be either ``A`` or ``B``, but we are looking for the subtype of ``A => B | C => D``. Let's walk through this...
```

Comment:
To be honest, I think this is a fun exercise for a blog post, but I'm worried that this example is too abstract and might discourage folks from grasping the concepts if introduced too early.